pystac_monty/sources/gfd.py

```diff
@@ -0,0 +1,204 @@
+            bbox=bbox,
+            datetime=startdate,
+            properties={
+                "title": description,
+                "description": description,
+                "start_datetime": startdate.isoformat(),
+                "end_datetime": enddate.isoformat(),
+            },
+        )
+
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = episode_number
+        monty.country_codes = data["cc"].split(",")
+        monty.hazard_codes = ["FL"]  # GFD is a Flood related source
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        return item
+
+    def make_hazard_items(self) -> List[Item]:
+        """Create hazard items"""
+        hazard_items = []
+
+        gfd_data = self.check_and_get_gfd_data()
+        event_items = self.make_source_event_items()
+
+        for event_item, src_data in zip(event_items, gfd_data):
+            hazard_item = event_item.clone()
+            hazard_item.id = event_item.id.replace(STAC_EVENT_ID_PREFIX, STAC_HAZARD_ID_PREFIX)
+            hazard_item.properties["roles"] = ["source", "hazard"]
+            hazard_item.set_collection(self.get_hazard_collection())
+
+            monty = MontyExtension.ext(hazard_item)
+            # Hazard Detail
+            monty.hazard_detail = HazardDetail(
+                cluster=self.hazard_profiles.get_cluster_code(hazard_item),
+                severity_value=src_data["dfo_severity"],
+                severity_unit="GFD Flood Severity Score",
+                severity_label=None,
+                estimate_type=MontyEstimateType.PRIMARY,
+            )
+            hazard_items.append(hazard_item)
+        return hazard_items
+
+    def make_impact_items(self) -> List[Item]:
+        """Create impact items"""
+        items = []
+        gfd_data = self.check_and_get_gfd_data()
+        event_items = self.make_source_event_items()
+        for event_item, src_data in zip(event_items, gfd_data):
+            items.extend(self.make_type_based_impact_items(event_item, src_data))
+        return items
+
+    def make_type_based_impact_items(self, event_item: dict, src_data: dict) -> List[Item]:
+        """Returns the impact details related to flood"""
+        impact_fields = {
+            "dfo_dead": (MontyImpactExposureCategory.ALL_PEOPLE, MontyImpactType.DEATH),
+            "dfo_displaced": (MontyImpactExposureCategory.ALL_PEOPLE, MontyImpactType.TOTAL_DISPLACED_PERSONS),
+        }
+
+        impact_items = []
+
+        for key_field, (category, impact_type) in impact_fields.items():
+            impact_item = event_item.clone()
+            impact_item.id = f"{STAC_IMPACT_ID_PREFIX}{src_data['system:index']} - {key_field}"
+            impact_item.properties["title"] = f"{event_item.properties['title']} - {key_field}"
```

Comment:
no white space in identifier
```suggestion
            impact_item.id = f"{STAC_IMPACT_ID_PREFIX}{src_data['system:index']}-{key_field}"
            impact_item.properties["title"] = f"{event_item.properties['title']}-{key_field}"
```

Reply:
Done